indicator/indicator.class.php

```diff
@@ -107,36 +108,63 @@ public function get_course_risks($startdate = null, $enddate = null) {
             }
         }
         $users = array_keys($users);
+		$this->userarray = $users;
 
         return self::get_risk_for_users($users, $startdate, $enddate);
     }
 
+	public function get_course_rawdata($startdate = null, $enddate = null) {
+		return $this->rawdata;
+	}
+	
+	public function get_course_users() {
+		return $this->userarray;
+	}
+	
     private function get_risk_for_users($userids, $startdate, $enddate) {
-        global $DB;
+        global $DB, $COURSE;
 
         if (empty($userids)) {
             return array();
         } else if (is_int($userids)) {
             $userids = array($userids);
         }
-
+		
+		// get query limit settings
+		$settingsnames = array('queryspecifydatetime', 'querystartdatetime', 'queryenddatetime');
+		$querysettings = new stdClass();
+		foreach ($settingsnames as $name) {
+			$tempvar = $DB->get_record_sql("SELECT * FROM {report_engagement} WHERE course = $COURSE->id AND indicator = '$name'");
```

Comment:
You really shouldn't be storing this kind of data in a new record using 'indicator' as your setting name.
Perhaps a new table is required to store these kind of settings for each course.

This is also make 3 queries to the database when it could be done in 1 - especially if the data was just in another table.